packages/editor/src/components/table-of-contents/index.js

```diff
@@ -30,8 +30,7 @@ function TableOfContents(
 			contentClassName="table-of-contents__popover"
 			renderToggle={ ( { isOpen, onToggle } ) => (
 				<Button
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```

Comment:
This looks good, thanks 👍 

Here are some details in case that can be useful:

`TableOfContents` is not in use since #44788.

When used somewhere, the button is an icon button, and it looks good with the new size:

| Before  | After |
| ------------- | ------------- |
| ![Screenshot 2024-09-11 at 15 05 49](https://github.com/user-attachments/assets/d16533c8-a1e7-4ef6-9281-07364c9ee5da)  | ![Screenshot 2024-09-11 at 15 06 55](https://github.com/user-attachments/assets/6cebf9bf-319b-45e2-bbcd-731c8611a40d)  |

No style overrides. However, the popover has some overrides (not something we should address in this PR of course), cc @WordPress/gutenberg-components:

https://github.com/WordPress/gutenberg/blob/8b1492131355aa6d42272faba25b841f0cd4509d/packages/editor/src/components/table-of-contents/style.scss#L1-L22

also cc @jorgefilipecosta should we deprecate `TableOfContents`?

Reply:
Hi @tyxla, yes I think we should deprecate it I will prepare a PR with the deprecation 👍

Reply:
Thanks @jorgefilipecosta, feel free to ping me for a review!